pinot-query-runtime/src/main/java/org/apache/pinot/query/service/dispatch/QueryDispatcher.java

```diff
@@ -204,6 +213,18 @@ void submit(long requestId, DispatchableSubPlan dispatchableSubPlan, long timeou
     }
   }
 
+  private static class StageInfo {
```

Comment:
do we still need stage info . I saw it got added during #12358 then removed in #12363  ?

Reply:
This is a simple helper class to help organize the info for each stage.